main.py

```diff
@@ -0,0 +1,353 @@
+                ship_explosion = Explosion(spaceship_x_position, spaceship_y_position, burst_size)
+                ship_explosions.append(ship_explosion)
+                loop_turns = 12
+            if boss_bullet_coll_spaceship is None:
+                pygame.display.update()
+            if loop_turns:
+                image_index, loop_times = self.draw_explosion(ship_explosions, image_index, loop_times)
+                loop_turns -= 1
+            if loop_turns == 1:
+                self.restart_game_menu()
+
+    def run_game(self):
+        pygame.init()
+        pygame.display.set_caption("Inwazja obcych")
+        bullets = []
+        aliens = []
+        alien_bullets = []
+        alien_explosions = []
+        ship_explosions = []
+        self.create_aliens(aliens)
+        loop_turns = 0
+        image_index = 0
+        loop_times = 0
+        loop_rotation_after_last_explosion = 0
+        self.ship.rect_start_x_position = 265
+        self.ship.rect_start_y_position = 680
+        ship_is_alive = True
+
+        while True:
+            pygame.time.Clock().tick(60)
+
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    sys.exit()
+                if event.type == pygame.KEYDOWN and len(bullets) < 5:
+                    if event.key == pygame.K_SPACE:
+                        bullets.append(SpaceshipBullet(self.ship.rect_start_x_position + ShipSettings.rect_width / 2.15,
+                                                       self.ship.rect_start_y_position))
+                        self.sound.play_laser_sound()
+
+            time_now = pygame.time.get_ticks()
+
+            GameSettings.screen.blit(GameSettings.background, (0, 0))
+
+            spaceship_position = self.ship.get_ship_rect(ship_is_alive)
+            spaceship_x_pos = spaceship_position[0]
+            spaceship_y_pos = spaceship_position[1]
+
+            self.bullet.get_bullet_rect(bullets)
+            self.bullet.change_bullet_position(bullets)
+
+            self.alien.get_alien_rect(aliens)
+            if self.alien.alien_movement(aliens):
+                self.restart_game_menu()
+
+            self.alien_bullet.get_alien_bullet_rect(alien_bullets)
+            self.alien_shot(time_now, aliens, alien_bullets)
+            self.alien_bullet.change_alien_bullet_position(alien_bullets)
+
+            alien_and_spaceship_bullet_coll = self.remove_alien_and_spaceship_bullet_if_collision(aliens, bullets)
+            self.explosion_if_collision(alien_and_spaceship_bullet_coll, alien_explosions)
+            image_index, loop_times = self.draw_explosion(alien_explosions, image_index, loop_times)
+
+            alien_bullet_coll_spaceship = self.alien_bullet_collision_spaceship(alien_bullets, spaceship_position)
+            alien_coll_spaceship = self.alien_collision_spaceship(aliens, spaceship_position)
+
+            if alien_bullet_coll_spaceship or alien_coll_spaceship is True:
```

Comment:
alien_coll_spaceship is True
to to samo co 
alien_coll_spaceship
Nie trzeba dawać is True